src/main/java/io/cdap/plugin/couchbase/sink/CouchbaseRecordWriter.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.couchbase.sink;
+
+import com.couchbase.client.java.Bucket;
+import com.couchbase.client.java.Cluster;
+import com.couchbase.client.java.CouchbaseCluster;
+import com.couchbase.client.java.document.JsonDocument;
+import com.google.common.base.Strings;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import io.cdap.plugin.couchbase.OperationType;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.RecordWriter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import rx.Observable;
+import rx.functions.Func1;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * {@link CouchbaseRecordWriter} writes the job outputs to the Couchbase. Accepts <code>null</code> key,
+ * {@link JsonDocument} pairs but writes only {@link JsonDocument} to the couchbase.
+ */
+public class CouchbaseRecordWriter extends RecordWriter<NullWritable, JsonDocument> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CouchbaseRecordWriter.class);
+  private static final Gson gson = new GsonBuilder().create();
+
+  private final Func1<JsonDocument, Observable<JsonDocument>> operation;
+  private final Cluster cluster;
+  private final Bucket bucket;
+  private final int batchSize;
+  private List<JsonDocument> batch;
+  private int totalCount;
+
+  public CouchbaseRecordWriter(TaskAttemptContext taskAttemptContext) {
```

Comment:
it's better to be more specific about what is passed into the constructor. In this case, pass in the CouchbaseSinkConfig instead of a TaskAttemptContext.

This would, for example, allow us to unit test the writer more easily.

Reply:
Changed to pass `CouchbaseSinkConfig` to the constructor.